src/js/video.js

```diff
@@ -96,10 +98,70 @@ function videojs(id, options, ready) {
   }
 
   // Element may have a player attr referring to an already created player instance.
-  // If not, set up a new player and return the instance.
-  return tag.player || Player.players[tag.playerId] || new Player(tag, options, ready);
+  // If so return that otherwise set up a new player below
+  if (tag.player || Player.players[tag.playerId]) {
+    return tag.player || Player.players[tag.playerId];
+  }
+
+  videojs.hooks('beforesetup').forEach(function(hookFunction) {
+    options = videojs.mergeOptions(options, hookFunction(tag, videojs.mergeOptions({}, options)));
```

Comment:
what happens if the `hookFunction` returns `null` or a string?
We probably should write some tests for this.

It may be easier to write tests if these are moved into a separate file and then copied over onto the videojs object but either way should work.